app/controllers/businessdetails/ConfirmRegisteredOfficeController.scala

```diff
@@ -37,15 +37,26 @@ class ConfirmRegisteredOfficeController @Inject () (val dataCache: DataCacheConn
                                                     val ds: CommonPlayDependencies,
                                                     val cc: MessagesControllerComponents) extends AmlsBaseController(ds, cc) {
 
-
   def updateBMAddress(bm: BusinessMatching): Option[RegisteredOffice] = {
-    bm.reviewDetails.fold[Option[RegisteredOffice]](None)(dtls => Some(RegisteredOfficeUK(
-      dtls.businessAddress.line_1,
-      dtls.businessAddress.line_2,
-      dtls.businessAddress.line_3,
-      dtls.businessAddress.line_4,
-      dtls.businessAddress.postcode.getOrElse("")
-    )))
+    bm.reviewDetails.fold[Option[RegisteredOffice]](None)(dtls =>
+      if(dtls.businessAddress.postcode.isDefined) {
+        Some(RegisteredOfficeUK(
+          dtls.businessAddress.line_1,
+          dtls.businessAddress.line_2,
+          dtls.businessAddress.line_3,
+          dtls.businessAddress.line_4,
+          dtls.businessAddress.postcode.getOrElse("")
+        ))
+      } else {
+        Some(RegisteredOfficeNonUK(
+          dtls.businessAddress.line_1,
+          dtls.businessAddress.line_2,
+          dtls.businessAddress.line_3,
+          dtls.businessAddress.line_4,
+          dtls.businessAddress.country
+        ))
+      }
+    )
```

Comment:
routing was extended to allow progress to the service for customers with non-UK country code, therefore this change to extend the registered office address to accept the non-UK one